### ISSUE TITLE
HTTPS rewritten to HTTP

### ISSUE THREAD

**Issue by flangam** [sidebase/nuxt-auth#374]
### Ask your question

I'm not sure if this is a bug.  In the callbackUrl, instead of using **https** as entered in the browser, it shows **http** instead.

![image](https://github.com/sidebase/nuxt-auth/assets/7723805/228a3854-ea84-4629-96aa-c9095f7c9e71)

This is how the application is started with PM2

![image](https://github.com/sidebase/nuxt-auth/assets/7723805/1776a7e8-64ba-44c1-87cf-67ab707ed1ed)

With this issue, authentication is not happening on the page being opened.

Here are the configuration:

![image](https://github.com/sidebase/nuxt-auth/assets/7723805/974c012e-f10e-4918-b82c-ae6fa9b38303)

![image](https://github.com/sidebase/nuxt-auth/assets/7723805/1e85ba4a-3b80-4a96-b2a4-06471644c0ca)

![image](https://github.com/sidebase/nuxt-auth/assets/7723805/5ef2c9ec-83a7-4d7c-9950-414f1f5de671)

**Comment by BracketJohn** [comment]
Hey @flangam 👋 

We'll need more to help you with this!

Can you please provide a working reproduction?

**Comment by ricricucit** [comment]
More info: this is only by activating: 

1. `@sidebase/nuxt-auth` in config
2. api endpoint (using `auth0` provider)
3. setting `env` vars


I have now tried also changing `basePath`, and also that gets ignored and I keep getting a `500 err` printing: `fetch failed (http://localhost:3000/api/auth/session?callbackUrl=http://localhost:3000/)` 🤷

**Comment by flangam** [comment]
I did the following:

- Added **NEXTAUTH_URL** in the env variables.  It's value is equal to **AUTH_ORIGIN**, if you haven't added that already.  Reference to this is in next-auth: https://next-auth.js.org/deployment
- Since we are using PM2, I created an ecosystem file (ecosystem.config.js) and defined all environment variables used for the production setup.

**Comment by ricricucit** [comment]
mh. First, thanks for your answer!

I did reach the same conclusions and already had `NEXTAUTH_URL` in the `.env`, with your hint, I've also added it to the nuxt `runtimeConfig`, just in case. 

...but still ❌ not working, still getting:

`[nuxt] [request error] [unhandled] [500] fetch failed (http://localhost:3000/api/auth/session?callbackUrl=http://localhost:3000/)`

💡 it seems to be only ignoring my `https` on some sort of "init", because when I switch to `http`, I correctly see the _Auth0 button_ and clicking on it: it correctly sends me to `https://URL`.

**Comment by ricricucit** [comment]
Oh, jeeee. ✅ solved.

[https://sidebase.io/nuxt-auth/configuration/nuxt-config](https://sidebase.io/nuxt-auth/configuration/nuxt-config) contains wrong infos:

❌  this
```js
...
    // The origin is set to the development origin. Change this when deploying to production by setting `origin` in this config before build-time or by exporting `AUTH_ORIGIN` by running `export AUTH_ORIGIN=...`
    origin: process.env.NEXTAUTH_URL,
...
```

✅  should be
```js
...
    // The origin is set to the development origin. Change this when deploying to production by setting `origin` in this config before build-time or by exporting `AUTH_ORIGIN` by running `export AUTH_ORIGIN=...`
    baseURL: process.env.NEXTAUTH_URL,
...
```

**Comment by ricricucit** [comment]
for the sidebase team and @sideborg: I'm referring to [this line](https://github.com/sidebase/docs/blob/main/content/2.nuxt-auth/2.configuration/2.nuxt-config.md?plain=1#L17) of your docs. 😅

**Comment by ricricucit** [comment]
for those that will end up here, to get straight asnwers, here's my ✅ *working Quick Start*:

`env`
```js
NEXTAUTH_URL=https://localhost:3000
AUTH_SECRET="442ANTe35tfg5!sd34tg"
AUTH_ORIGIN="https://localhost:3000"
```

`nuxt.config.ts`
```js
runtimeConfig: {
    public: {
        ...
    },
    NEXTAUTH_URL: process.env.NEXTAUTH_URL,
    AUTH_SECRET: process.env.AUTH_SECRET,
    AUTH_ORIGIN: process.env.AUTH_ORIGIN
    ...
},
```


```js
auth: {
    // The module is enabled. Change this to disable the module
    isEnabled: true,
    // The origin is set to the development origin. Change this when deploying to production by setting `origin` in this config before build-time or by exporting `AUTH_ORIGIN` by running `export AUTH_ORIGIN=...`
    // origin: process.env.NEXTAUTH_URL, // leaving this, as Docs say it SHOULD be
    baseURL: process.env.AUTH_ORIGIN,
    // The base path to the authentication endpoints. Change this if you want to add your auth-endpoints at a non-default location
    basePath: '/api/auth',
    // Whether to periodically refresh the session. Change this to `true` for a refresh every seconds or set this to a number like `5000` for a refresh every 5000 milliseconds (aka: 5 seconds)
    enableSessionRefreshPeriodically: false,
    // Whether to refresh the session whenever a window focus event happens, i.e, when your user refocuses the window. Set this to `false` to turn this off
    enableSessionRefreshOnWindowFocus: false,
    // Whether to add a global authentication middleware that will protect all pages without exclusion
    globalAppMiddleware: false,
    // Select the default-provider to use when `signIn` is called. Setting this here will also effect the global middleware behavior: E.g., when you set it to `github` and the user is unauthorized, they will be directly forwarded to the Github OAuth page instead of seeing the app-login page
    defaultProvider: 'auth0',
    // Configuration of the global auth-middleware (only applies if you set `globalAppMiddleware: true` above!)
    globalMiddlewareOptions: {
      // Whether to allow access to 404 pages without authentication. Set this to `false` to force users to sign-in before seeing `404` pages. Setting this to false may lead to vue-router problems (as the target page does not exist)
      allow404WithoutAuth: true,
      // Whether to automatically set the callback url to the page the user tried to visit when the middleware stopped them. This is useful to disable this when using the credentials provider, as it does not allow a `callbackUrl`. Setting this to a string-value will result in that being used as the callbackUrl path. Note: You also need to set the global `addDefaultCallbackUrl` setting to `false` if you want to fully disable this for the global middleware.
      addDefaultCallbackUrl: false
    }
```

`/server/api/[...].ts`

```ts
import { NuxtAuthHandler } from '#auth'
import Auth0Provider from 'next-auth/providers/auth0'

export default NuxtAuthHandler({
    secret: useRuntimeConfig().AUTH_SECRET,
    origin: useRuntimeConfig().AUTH_ORIGIN,
    providers: [
        // @ts-expect-error You need to use .default here for it to work during SSR. May be fixed via Vite at some point
        Auth0Provider.default({
          clientId: useRuntimeConfig().AUTH0_CLIENT_ID_APP,
          clientSecret: useRuntimeConfig().AUTH0_CLIENT_SECRET_APP,
          issuer: 'https://' + useRuntimeConfig().AUTH0_DOMAIN_APP
        })
    ]
})
```

**Comment by ricricucit** [comment]
go ahead 🙂 

Actually: I have updated my asnwer, in the `nuxt.config.ts` part, as I noticed that also there, I needed to have `baseURL: process.env.AUTH_ORIGIN,`.............BUT: I'm not sure if `origin: process.env.AUTH_ORIGIN,` is also used.

**Comment by andreasvirkus** [comment]
@BracketJohn this is still not fixed in the docs (only in 0.6 beta docs) 😞
